src/ui/frmpreferences.cpp

```diff
@@ -289,42 +252,63 @@ void frmPreferences::loadTranslations(QSettings *s)
                     QLocale::languageToString(QLocale(localizationSetting).language()), Qt::DisplayRole));
 }
 
-void frmPreferences::saveLanguages(QSettings *s)
+void frmPreferences::saveTranslation()
 {
-    QString keyPrefix = "Languages/";
+    QMap<QString, QVariant> selected = ui->localizationComboBox->currentData().toMap();
+    m_settings.General.setLocalization(selected.value("langCode").toString());
+}
 
-    for (QVariantMap::iterator prop = m_commonLanguageProperties->begin(); prop != m_commonLanguageProperties->end(); ++prop) {
-        s->setValue(keyPrefix + prop.key(), m_langsTempSettings->value(keyPrefix + prop.key()));
+void frmPreferences::loadShortcuts()
+{
+    MainWindow* mw = qobject_cast<MainWindow*>(parent());
+
+    auto&& allActions = mw->getActions();
+
+    //getActions() also returns actions like separators or menu entries that don't have any
+    //shortcuts. We can remove them based on their objectName and iconText.
+    for(auto&& action : allActions) {
+        if(!action->objectName().isEmpty() && !action->iconText().isEmpty())
+            m_actions[action->objectName()] = action;
     }
 
-    for (int i = 0; i < m_langs.length(); i++) {
-        const QMap<QString, QString> &map = m_langs.at(i);
+    m_keyGrabber = new KeyGrabber();
 
-        QString langId = map.value("id", "");
-        QString keyPrefix = "Languages/" + langId + "/";
 
-        s->setValue(keyPrefix + "useDefaultSettings", m_langsTempSettings->value(keyPrefix + "useDefaultSettings"));
+    //Build the interface
+    QWidget *container = ui->page_5;
+    QVBoxLayout *layout = new QVBoxLayout();
+    QPushButton *resetDefaults = new QPushButton("Restore Defaults");
+    QObject::connect(resetDefaults,SIGNAL(clicked()),this,SLOT(resetShortcuts()));
+    resetDefaults->setFixedWidth(128);
```

Comment:
This is actually code that was already in there. You'll find it again in the file a little further up, where it was "deleted".
I actually would like the entire block to go away. All it does is create the contents of the Shortcuts page, but the layout and button could just be added in the designer instead. I just couldn't figure out how to. The StackedWidget is really stubborn in the designer. I think that's what the original author of the code thought too, which is why they just added the widgets this way.

Anyways, I'll change the connect call.